docs/site/Loopback-component-authorization.md

```diff
@@ -8,543 +8,227 @@ permalink: /doc/en/lb4/Loopback-component-authorization.html
-        // Remove permission if it is disallowed for user
-        perms.splice(perms.indexOf(userPerm.permission), 1);
-      }
-    });
-    return perms;
+  @authorize.skip()
+  @get('/hello')
+  hello(): string {
+    return 'Hello';
   }
 }
 ```
 
-Next, we need to expose these providers via Component to be bound to the
-context.
-
-{% include code-caption.html content="/src/authorization/component.ts" %}
+## Programming Access Policies
 
-```ts
-import {Component, ProviderMap} from '@loopback/core';
-import {AuthorizationBindings} from './keys';
-import {AuthorizeActionProvider} from './providers/authorization-action.provider';
-import {AuthorizationMetadataProvider} from './providers/authorization-metadata.provider';
-import {UserPermissionsProvider} from './providers/user-permissions.provider';
-
-export class AuthorizationComponent implements Component {
-  providers?: ProviderMap;
-
-  constructor() {
-    this.providers = {
-      [AuthorizationBindings.AUTHORIZE_ACTION.key]: AuthorizeActionProvider,
-      [AuthorizationBindings.METADATA.key]: AuthorizationMetadataProvider,
-      [AuthorizationBindings.USER_PERMISSIONS.key]: UserPermissionsProvider,
-    };
-  }
-}
-```
+Users are expected to program policies that enforce access control, in classes
+of type `Authorizer` as below. The `authorize()` function in this class is
+expected to be called by the `Authorization Interceptor` which is called for
+every API endpoint decorated with `@authorize()`.
 
-You can see that we have used the same binding keys which we created earlier.
-
-Now, its time to create our method decorator function. Here it is. We will be
-using the same metadata accessor key which we created earlier and the metadata
-interface for accessing the data in decorator.
-
-{% include code-caption.html content="/src/authorization/decorators/authorize.decorator.ts" %}
+> Usually the `authorize()` function in a `Authorizer` is binded through a
+> provider as below
 
 ```ts
-import {MethodDecoratorFactory} from '@loopback/core';
-import {AuthorizationMetadata} from '../types';
-import {AUTHORIZATION_METADATA_ACCESSOR} from '../keys';
-
-export function authorize(permissions: string[]) {
-  return MethodDecoratorFactory.createDecorator<AuthorizationMetadata>(
-    AUTHORIZATION_METADATA_ACCESSOR,
-    {
-      permissions: permissions || [],
-    },
-  );
-}
-```
-
-For error handling keys, lets create an enum.
-
-{% include code-caption.html content="/src/authorization/error-keys.ts" %}
+class MyAuthorizationProvider implements Provider<Authorizer> {
+  /**
+   * @returns authenticateFn
+   */
+  value(): Authorizer {
+    return this.authorize.bind(this);
+  }
 
-```ts
-export const enum AuthorizeErrorKeys {
-  NotAllowedAccess = 'Not Allowed Access',
+  async authorize(
+    context: AuthorizationContext,
+    metadata: AuthorizationMetadata,
+  ) {
+    events.push(context.resource);
+    if (
+      context.resource === 'OrderController.prototype.cancelOrder' &&
```

Comment:
Can you also give an example how the authorize options mentioned earlier

```
@authorize({resource: 'order', scopes: ['create']})
``` 

would be handled here?

Reply:
agreed